javalib/src/main/scala/java/math/MathContext.scala

```diff
@@ -0,0 +1,206 @@
+/*
+ * Ported by Alistair Johnson from https://android.googlesource.com/platform/libcore/+/master/luni/src/main/java/java/math/MathContext.java
+ * Original license copied below:
+*/
+
+/*
+ *  Licensed to the Apache Software Foundation (ASF) under one or more
+ *  contributor license agreements.  See the NOTICE file distributed with
+ *  this work for additional information regarding copyright ownership.
+ *  The ASF licenses this file to You under the Apache License, Version 2.0
+ *  (the "License"); you may not use this file except in compliance with
+ *  the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package java.math
+
+import java.io.ObjectInputStream
+import java.io.Serializable
+
+object MathContext {
+
+  /**
+   * A {@code MathContext} which corresponds to the <a href="http://en.wikipedia.org/wiki/IEEE_754-1985">IEEE 754</a> quadruple
+   * decimal precision format: 34 digit precision and
+   * {@link RoundingMode#HALF_EVEN} rounding.
+   */
+  val DECIMAL128 = MathContext(34, RoundingMode.HALF_EVEN)
+
+  /**
+   * A {@code MathContext} which corresponds to the <a href="http://en.wikipedia.org/wiki/IEEE_754-1985">IEEE 754</a> single decimal
+   * precision format: 7 digit precision and {@link RoundingMode#HALF_EVEN}
+   * rounding.
+   */
+  val DECIMAL32 = MathContext(7, RoundingMode.HALF_EVEN)
+
+  /**
+   * A {@code MathContext} which corresponds to the <a href="http://en.wikipedia.org/wiki/IEEE_754-1985">IEEE 754</a> double decimal
+   * precision format: 16 digit precision and {@link RoundingMode#HALF_EVEN}
+   * rounding.
+   */
+  val DECIMAL64 = MathContext(16, RoundingMode.HALF_EVEN)
+
+  /**
+   * A {@code MathContext} for unlimited precision with
+   * {@link RoundingMode#HALF_UP} rounding.
+   */
+  val UNLIMITED = MathContext(0, RoundingMode.HALF_UP)
+
+  private def getArgs(s: String): (Int, RoundingMode) = {
+    checkNotNull(s, "null string")
+    val precisionLength: Int = "precision=".length
+    val roundingModeLength: Int = "roundingMode=".length
+    val spaceIndex: Int = s.indexOf(' ', precisionLength)
+
+    if (!s.startsWith("precision=") || spaceIndex == -1) {
+      throw invalidMathContext("Missing precision", s)
+    }
+    val precisionString = s.substring(precisionLength, spaceIndex)
+
+    val precision = try {
+        java.lang.Integer.parseInt(precisionString)
+      }
+      catch {
+        case _: Throwable => throw invalidMathContext("Bad precision", s)
+      }
```

Comment:
Indentation issue.
Plus, put `catch` on the same line as the `}`.
Plus, don't catch `Throwable`. Catch `NumberFormatException` instead.
Plus, remove the `throw`, since `invalidMathContext` throws itself.